bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4347,10 +4347,15 @@ ReplicaImp::ReplicaImp(bool firstTime,
       auto timeSinceLastSnapshot = (std::chrono::duration_cast<std::chrono::seconds>(currTime) -
                                     DbCheckpointManager::instance().getLastCheckpointCreationTime())
                                        .count();
-      if (getReplicaConfig().dbCheckpointFeatureEnabled && seqNum && isCurrentPrimary() &&
-          !(seqNum % getReplicaConfig().dbCheckPointWindowSize) &&
-          (timeSinceLastSnapshot >= getReplicaConfig().dbSnapshotIntervalSeconds.count())) {
-        DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(seqNum);
+      if ((getReplicaConfig().dbCheckpointFeatureEnabled && seqNum && isCurrentPrimary() &&
+           !(seqNum % getReplicaConfig().dbCheckPointWindowSize) &&
+           (timeSinceLastSnapshot >= getReplicaConfig().dbSnapshotIntervalSeconds.count())) ||
+          (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().has_value() &&
+           DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().value() == seqNum)) {
+        DbCheckpointManager::instance().sendInternalCreateDbCheckpointMsg(seqNum, false);
+        if (DbCheckpointManager::instance().getNextStableSeqNumToCreateSnapshot().has_value()) {
```

Comment:
we don't need this change. See the comment below

Reply:
Fixed